airflow/models/trigger.py

```diff
@@ -87,7 +89,15 @@ def bulk_fetch(cls, ids: Iterable[int], session=None) -> dict[int, Trigger]:
         Fetches all of the Triggers by ID and returns a dict mapping
         ID -> Trigger instance
         """
-        return {obj.id: obj for obj in session.query(cls).filter(cls.id.in_(ids)).all()}
+        return {
+            obj.id: obj
+            for obj in session.query(cls)
+            .filter(cls.id.in_(ids))
+            .options(joinedload("task_instance"))
+            .options(joinedload("task_instance.trigger"))
+            .options(joinedload("task_instance.trigger.triggerer_job"))
```

Comment:
Do we only need these for trigger log, and is it possible to not do these for non-trigger log? (And does it make much difference in performance? I’m not even sure about that.)

Reply:
yeah we could do.. i guess the one thing i was fuzzy about is... does it make it an inner join -- here it doesn't matter since the rows are all there... but  i guess we add `isouter=True` and should work i can try if you think good idea.

but i guess it would be marginally more costly ... not sure how much